### PR TITLE
darkmode

### PR DESCRIPTION
**Pull request by nsnave**
Used darkreader library for easy conversion of site (https://github.com/darkreader/darkreader)

### CODE REVIEW

**Comment by hsheth2** [comment]
dang that's sick

File: frontend/src/components/Navbar.js [hunk 1]
```diff
@@ -46,6 +53,32 @@ function CourseTableNavbar({ isLoggedIn }) {
     window.location.pathname = '/';
   };
 
+  // DarkMode or LightMode
+  setFetchMethod(window.fetch);
+
+  // Checks user preferences for dark mode
+  if (!window.localStorage.getItem('darkmode')) {
+    var temp =
```
Comment (hsheth2) [inline frontend/src/components/Navbar.js:61]: Generally with JS we want to use `let` or `const` - there's a bunch of weird scoping issues with `var`

File: frontend/src/components/Navbar.js [hunk 2]
```diff
@@ -46,6 +53,32 @@ function CourseTableNavbar({ isLoggedIn }) {
     window.location.pathname = '/';
   };
 
+  // DarkMode or LightMode
+  setFetchMethod(window.fetch);
```
Comment (hsheth2) [inline frontend/src/components/Navbar.js:57]: what does this do?

File: frontend/src/components/DarkModeButton.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+/**
+ * DarkMode Button
+ * @prop darkModeEnabled - boolean to determine which image to display (true for "sun"; false for "moon")
+ */
+
+function DarkModeButton({ darkModeEnabled = false }) {
+  return (
+    <span>
+      <img
+        src={darkModeEnabled ? sun : moon}
```
Comment (hsheth2) [inline frontend/src/components/DarkModeButton.js:17]: We already have the react-icons library and we're mostly using icons from font-awesome.

The FaSun and FaMoon/FaRegMoon might be a good fit, and then we can avoid the extra png's

**Comment by maxyuan6717** [comment]
Thoughts on sticking the toggle under the me dropdown?

**Comment by hsheth2** [comment]
I kinda like it in the navbar - it's much more discoverable there. It also reminds me of the interface in umami, which I think is pretty clean.

File: frontend/src/components/Navbar.js (hunk 2 above)
Comment (nsnave) [reply frontend/src/components/Navbar.js:57]: not 100% sure on all the details but sometimes the darkreader library needs to access "a cross-origin resource" and without this line it throws an error

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**